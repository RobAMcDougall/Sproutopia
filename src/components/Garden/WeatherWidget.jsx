import React, { useState } from 'react';
import '../../css/weather.css'; 

const WeatherWidget = () => {
  const [postcode, setPostcode] = useState('');
  const [weatherData, setWeatherData] = useState(null);
  const API_KEY = 'e48e17fa5e894047ac8161125242103';
  const API_URL = `https://api.weatherapi.com/v1/current.json?key=${API_KEY}&q=${postcode}`;

  const handleSubmit = async (e) => {
    e.preventDefault();
    try {
      const response = await fetch(API_URL);
      const data = await response.json();
      setWeatherData(data);
    } catch (error) {
      console.error('Error fetching weather data:', error);
    }
  };

  return (
    <div className="weather-widget">
      <form onSubmit={handleSubmit}>
        <label>
          Enter your postcode:
          <input
            type="text"
            value={postcode}
            onChange={(e) => setPostcode(e.target.value)}
          />
        </label>
        <button type="submit">Get Weather</button>
      </form>
      {weatherData && (
        <div>
          <h2>Weather Information</h2>
          <p>Location: {weatherData.location.name}</p>
          <p>Temperature: {weatherData.current.temp_c}Â°C</p>
          <p>Condition: {weatherData.current.condition.text}</p>
          {/* Display weather condition icon */}
          <img src={weatherData.current.condition.icon} alt="Weather Icon" />
          {/* Check if it's raining and render the message accordingly */}
          {weatherData.current.condition.text.toLowerCase().includes('rain') && (
            <p className="rain-message">Looks like it might be raining where you are. You may not need to water your plants today.</p>
          )}
          {/* Add more weather information as needed */}
        </div>
      )}
    </div>
  );
};

export default WeatherWidget;