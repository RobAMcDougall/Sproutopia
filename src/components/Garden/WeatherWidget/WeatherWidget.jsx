import React, { useState } from 'react';
import '../../../css/weather.css'; 

const WeatherWidget = () => {
  const [postcode, setPostcode] = useState('');
  const [weatherData, setWeatherData] = useState(null);
  const API_KEY = 'e48e17fa5e894047ac8161125242103';
  const API_URL = `https://api.weatherapi.com/v1/current.json?key=${API_KEY}&q=${postcode}`;

  const handleSubmit = async (e) => {
    e.preventDefault();
    try {
      const response = await fetch(API_URL);
      const data = await response.json();
      setWeatherData(data);
    } catch (error) {
      console.error('Error fetching weather data:', error);
    }
  };

  return (
    <div className="weather-widget">
      <form onSubmit={handleSubmit}>
        <label>
          Enter your postcode to get the latest weather information:
          <input
            type="text"
            value={postcode}
            onChange={(e) => setPostcode(e.target.value)}
          />
        </label>
        <button type="submit">Get Weather</button>
      </form>
      {weatherData && (
        <div>
          <p>Location: {weatherData.location.name}</p>
          <p>Temperature: {weatherData.current.temp_c}Â°C</p>
          <p>Condition: {weatherData.current.condition.text}</p>
          <img src={weatherData.current.condition.icon} alt="Weather Icon" />
          {weatherData.current.condition.text.toLowerCase().includes('rain') && (
            <p className="rain-message">Looks like it might be raining where you are. You may not need to water your plants today.</p>
          )}
        </div>
      )}
    </div>
  );
};

export default WeatherWidget;