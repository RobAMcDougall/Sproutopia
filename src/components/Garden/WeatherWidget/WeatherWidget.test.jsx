import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';

import WeatherWidget from './WeatherWidget';

describe('WeatherWidget', () => {
  it('renders correctly', () => {
    render(<WeatherWidget />);
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<WeatherWidget />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('updates postcode state when typing in input field', () => {
    render(<WeatherWidget />);
    const input = screen.getByLabelText('Enter your postcode to get the latest weather information:');
    fireEvent.change(input, { target: { value: '12345' } });
    expect(input.value).toBe('12345');
  });
})