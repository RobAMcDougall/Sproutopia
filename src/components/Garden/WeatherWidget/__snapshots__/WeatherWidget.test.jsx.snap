// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`WeatherWidget > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="weather-widget"
  >
    <form>
      <label>
        Enter your postcode to get the latest weather information:
        <input
          type="text"
          value=""
        />
      </label>
      <button
        type="submit"
      >
        Get Weather
      </button>
    </form>
  </div>
</DocumentFragment>
`;
